### DISCUSSION TITLE
Is it possible to use a single VPS for multiple OMR installations?

### DISCUSSION THREAD

**Discussion by mzdrahal** [Ysurac/openmptcprouter#3543] [Q&A] [answered]
Is the OMR <-> VPS link exclusive, meaning do I need a separate VPS for every OMR installation or can I use a single VPS for multiple OMR clients?

**Comment by Ysurac** [comment, marked as answer]
It's possible to use one VPS for multiple installations, but not documented and I don't want to manage support on this.

**Reply by sebastien76370** [reply]
Thank for you answer.
we're creating mutli VM  on vps ubuntu .
Each VM will have a configuration  with a list of different port (ssh ...)
Do you think , it will work please ?
Thank for your help

**Reply by itsgps** [reply]
From someone not using the product much yet, but having a significant amount of experience in networking and routing - should be no reason it wouldn't work...But with the Range of Ports Ysurac has allocated to services...you are somewhat limited to the number of VPS you might want to run (if on a single IP Address.)
So - then it comes down to how you are able to serve the VPS up to the Public IPs - i.e. if using only 1 Public facing IP, you need to change Ports... But if you somehow are able to get Multiple IPs from your ISP - then EASY PEASY, just add the same NAT'ing to each seperate IP Address.  Going back 20 or so years, getting Multiple IPs from an ISP was Dead simple, then you had to Justify the reasons...Then soon enough you are lucky to even get a Single Static IP through asking, and even then you have to pay for it!

I'll be in the same boat for some future implementations, because I want to have 4 or 5 Portable units out in the field, and all congregating for Egress to our Office Fibre connection...But Either I need to convince the ISP I need more IP addresses, or I need to setup each of them with different Port ranges

**Comment by yannick2801** [comment]
It would be interesting to have your perspective on how to set up multiple mptcpRouters on a single VPS. I understand that you may not offer support for this, but if you could guide us on the best way to proceed, I would be grateful. We will keep you updated on our progress. Best regards

**Comment by sebastien76370** [comment]
Hello,

I’m still trying to create VMs on a VPS (using VirtualBox), with each VM acting as a unique router.

The principle is that each VM includes an installation of OpenMPTCPRouter using the following command:

wget -O - https://www.openmptcprouter.com/server/debian-x86_64.sh | KERNEL="5.4" sh
The creation of the VMs worked perfectly.

To make each VM unique, I set up port forwarding from the VM to OpenMPTCPRouter.
The forwarded ports are:

SSH: 65222 (TCP)
Shadowsocks: 65101 (TCP & UDP)
Shadowsocks-Rust: 65280 (TCP & UDP)
Glorytun: 65001 (TCP & UDP)
OMR JSON Admin: 65500 (TCP)
OpenVPN: 65301 (TCP)
MLVPN: 65201–65208 (UDP)
iPerf3: 65400 (TCP & UDP)
DSVPN: 65401 (TCP)
V2Ray: 65228, 65229, 65230, 65231 (TCP)
Xray: 65248, 65249, 65250, 65251, 65252, and 443 (when VLESS-Reality is used) (TCP)
WireGuard: 65311, 65312 (UDP)
The idea is to forward the ports as follows:

61xxxx (VM) -> 65xxxx for VM1
62xxxx (VM) -> 65xxxx for VM2
63xxxx (VM) -> 65xxxx for VM3
64xxxx (VM) -> 65xxxx for VM4
65xxxx (VM) -> 65xxxx for VM5
On the router, I modified all the ports in the configuration files located in /etc/config to point to 61xxxx for VM1, 62xxxx for VM2, and so on.

Changed configuration files:
dhcp, https-dns-proxy, openmptcprouter, sysupgrade
dropbear, igmpproxy, openvpn, ttyd
dscp, iperf, openvpn_recipes, ucitrack
dsvpn, luci, packet_capture, uhttpd
etherwake, mail, rpcd, unbound
firewall, mlvpn, selhagroup, upnpd
fstab, network, shadowsocks-libev, v2ray
glorytun, omr-bypass, snmpd, vnstat
glorytun-opkg, omr-quota, sqm, wireless
glorytun-udp, omr-tracker, system
Current Issues:
Router Configuration Reset:
The ports I modified on the router revert to their original values. For example:

Replacing 65001 with 64001 in the glorytun configuration:

config glorytun 'vpn'
  option port '64001'
After some time, the system resets 64001 back to 65001.
Script Errors on the VM:
The /usr/bin/omr-admin script on the VM does not start because of two errors:

Missing Parentheses in if ipv6_enabled:

if ipv6_enabled:
Should be:


if ipv6_enabled():
Typo in the ipv6_enabled Function:

addr = netifaces.ifaddresses('lo')  # Incorrect
Should be:


addrs = netifaces.ifaddresses('lo')  # Correct
Updated function:


def ipv6_enabled():
    ipv6_enabled = False
    addrs = netifaces.ifaddresses('lo')
    ipv4_addr_list = addrs[netifaces.AF_INET]
    for ip_info in ipv4_addr_list:
        addr = ip_info['addr']
        if IPAddress(addr).version == 6:
            ipv6_enabled = True
    return ipv6_enabled
Help Needed:
How can I make the router keep the updated ports (e.g., 61xxxx for VM1) instead of resetting them to the defaults?
A reliable process for remapping the default port range (65xxxx) to 64xxxx or any other custom range.
Thank you for your assistance!

**Reply by q2go** [reply]
Well, if you made it work by editing all those files, you could also come up with a set of `sed` command lines which apply the same changes to each file. And these `sed` scripts could include a shell variable to specify the port number prefix.
